### PR TITLE
Move sanitizers test to cmake

### PR DESCRIPTION
**Pull request by clementperon**

### CODE REVIEW

**Review by egecetin — APPROVED** [review]

File: .cirrus.yml [hunk 1]
```diff
@@ -101,30 +104,31 @@ task:
...
+    - wget https://www.tcpdump.org/release/libpcap-1.10.3.tar.gz && tar -xvf libpcap-1.10.3.tar.gz && cd libpcap-1.10.3 && cmake -DBUILD_SHARED_LIBS=OFF -S . -B build && cmake --build build -j 2 &&  cmake --install build && cd ..
+    - cmake -S . -B $BUILD_DIR -DPCAPPP_USE_SANITIZER=Fuzzer -DPCAPPP_BUILD_FUZZERS=ON -DPCAPPP_BUILD_EXAMPLES=OFF
   build_script:
-    - export LIB_FUZZING_ENGINE="-fsanitize=fuzzer"
-    - make fuzzers
+    - cmake --build $BUILD_DIR -j 2
   test_script:
-    - cd Tests/Fuzzers/RegressionTests && ./run_tests.sh
+    # Run the fuzzer for 10 minutes
+    - $BUILD_DIR/Tests/Fuzzers/Fuzzer -max_total_time=600 -max_len=65535 -workers=2
```
Comment (seladb) [inline .cirrus.yml:118]: No need to run the fuzzer, it's already running in [Google OSS Fuzz](https://google.github.io/oss-fuzz/). In the CI we only run the regression tests which rerun issues that were found and fixed to make sure they don't appear again

File: .cirrus.yml [hunk 2]
```diff
@@ -101,30 +104,31 @@ task:
...
-    - make fuzzers
+    - cmake --build $BUILD_DIR -j 2
   test_script:
-    - cd Tests/Fuzzers/RegressionTests && ./run_tests.sh
+    - python3.8 -m pip install -U pip
+    - python3.8 -m pip install -r ci/run_tests/requirements.txt
+    - python3.8 ci/run_tests/run_tests.py --interface eth0
 
 task:
   name: Regression Tests Linux (Undefined Behavior Sanitizer)
```
Comment (seladb) [inline .cirrus.yml:134]: We also want to move the Undefined Behavior Sanitizer to CMake, no? 🤔

File: Tests/Fuzzers/FuzzTarget.cpp [hunk 3]
```diff
@@ -40,7 +40,7 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
 	if (dumpDataToPcapFile(Data, Size) < 0)
 	{
 		std::cerr << "Can't Dump buffer to a PCAP file!!!!\n";
-		return -1;
+		return 0;
```
Comment (seladb) [inline Tests/Fuzzers/FuzzTarget.cpp:43]: Why changing the return value to `0`?

File: .cirrus.yml (hunk 1 above)
Comment (clementperon) [reply .cirrus.yml]: fixed

File: Tests/Fuzzers/FuzzTarget.cpp (hunk 3 above)
Comment (clementperon) [reply Tests/Fuzzers/FuzzTarget.cpp:43]: I misunderstood even if the data is not valid we should add the pattern to the corpus.
see: https://github.com/the-tcpdump-group/libpcap/blob/master/testprogs/fuzz/fuzz_pcap.c

File: .cirrus.yml (hunk 2 above)
Comment (clementperon) [reply .cirrus.yml:135]: done

File: Tests/Fuzzers/FuzzTarget.cpp [hunk 4]
```diff
@@ -51,15 +51,19 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
 	if (!reader.open())
 	{
 		std::cerr << "Error opening the pcap file\n";
-		return -1;
+		return 0;
 	}
 
+	for (size_t i = 0; i < Size; i++)
+		printf("0x%02X ", Data[i]);
+	printf("\n\n");
```
Comment (seladb) [inline Tests/Fuzzers/FuzzTarget.cpp:59]: I guess it's for debug purposes? Can we remove it?

File: .cirrus.yml [hunk 5]
```diff
@@ -101,42 +104,47 @@ task:
...
+#     image: gcr.io/oss-fuzz-base/base-clang
+#   configure_script:
+#     - apt-get update && apt-get install -y cmake git libpcap-dev
+#     - cmake -S . -B $BUILD_DIR -DPCAPPP_USE_SANITIZER=MemorySanitizer -DPCAPPP_BUILD_FUZZERS=ON -DPCAPPP_BUILD_EXAMPLES=OFF
+#   build_script:
+#     - cmake --build $BUILD_DIR -j 2
+#   test_script:
+#     - export BINARY="$BUILD_DIR/Tests/Fuzzers/FuzzTarget"
+#     - export SAMPLES="Tests/Fuzzers/RegressionTests/regression_samples"
+#     - Tests/Fuzzers/RegressionTests/run_tests.sh
```
Comment (seladb) [inline .cirrus.yml:133]: I think we can merge the PR as is and handle that later

**Review by seladb — APPROVED** [review]
Please see 1 comment, I think we can fix it and merge the PR!

File: Tests/Fuzzers/FuzzTarget.cpp (hunk 4 above)
Comment (clementperon) [reply Tests/Fuzzers/FuzzTarget.cpp]: done

**Pull request merged**